### DISCUSSION TITLE
Verify 17.9 has a breaking change

### DISCUSSION THREAD

**Discussion by 304NotModified** [VerifyTests/Verify#591] [General]
Unfortunately Verify 17.9 has a breaking change. 

The dependency of DiffEngine is moved from 9 to 10.  

This works in 17.8.x and breaks (binary and on build) with 17.9:

```c#
DiffTools.AddTool(
    name: name,
    exePath: vsDiffPath,
    targetLeftArguments: launchArguments,
    targetRightArguments: launchArguments
    requiresTarget: true, autoRefresh: true, isMdi: true, supportsText: true, binaryExtensions: new List<string>()
);
```

Will give: 

> Error CS1739: The best overload for 'AddTool' does not have a parameter named 'targetLeftArguments'

I assumed this library is following  [semver](https://semver.org/). If so. please make only breaking changes in Major versions :)

**Comment by SimonCropp** [comment]
the API of Verify did not change. so i changed the minor. 

This should work for you

```
        DiffTools.AddTool(
            name: "name",
            exePath: "vsDiffPath",
            new LaunchArguments(
                Left: launchArguments,
                Right: launchArguments),
            requiresTarget: true, autoRefresh: true, isMdi: true, supportsText: true, binaryExtensions: new List<string>()
        );
```

**Reply by 304NotModified** [reply]
Yes but it's still a breaking change. 

I prefer floating version (e.g 17.*), but then we need to be sure there are no breaking changes. 

Thanks for the new code 👍
